sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/utils/ApplicationId.java

```diff
@@ -26,20 +33,29 @@
     //    cos: for Cosmos
     //    aad: for AAD
     //    b2c: for AAD B2C
-    public static final String VERSION = Optional.of(ApplicationId.class)
-                                                 .map(Class::getPackage)
-                                                 .map(Package::getImplementationVersion)
-                                                 .orElse("unknown");
+    private static final Logger LOGGER = LoggerFactory.getLogger(ApplicationId.class);
+    public static final String VERSION = getVersion();
     public static final String AZURE_SPRING_KEY_VAULT = "az-sp-kv/" + VERSION;
     public static final String AZURE_SPRING_SERVICE_BUS = "az-sp-bus/" + VERSION;
     public static final String AZURE_SPRING_STORAGE_BLOB = "az-sp-sb/" + VERSION;
     public static final String AZURE_SPRING_STORAGE_FILES = "az-sp-sf/" + VERSION;
     /**
-     * AZURE_SPRING_AAD does not contain VERSION, because AAD server support 2 headers:
-     * 1. x-client-SKU;
-     * 2. x-client-VER;
+     * AZURE_SPRING_AAD does not contain VERSION, because AAD server support 2 headers: 1. x-client-SKU; 2.
+     * x-client-VER;
      */
     public static final String AZURE_SPRING_AAD = "az-sp-aad";
     public static final String AZURE_SPRING_B2C = "az-sp-b2c";
 
+    private static String getVersion() {
+        String version = "unknown";
+        Properties properties = new Properties();
+        try (InputStream inputStream =
```

Comment:
spring has api to do this thing, we should leverage that instead of dealing with these stream by ourselves ( I understand this is easy, but still)

`org.springframework.boot.autoconfigure.info.ProjectInfoAutoConfiguration#loadSource`

Reply:
Updated.

I didn't use `org.springframework.boot.autoconfigure.info.ProjectInfoAutoConfiguration#loadSource` because it protected. I use `PropertiesLoaderUtils` and `ClassPathResource` instead.

Reply:
yes, that was just an usage example for you